proposed/disable-multi-level-lookup-by-default.md

```diff
@@ -0,0 +1,204 @@
+# Disable multi-level lookup by default
+
+## Multi-level lookup
+
+Running framework dependent application or SDK command performs a search for
+install location of .NET (.NET Core or .NET 5+). If the user has private install
+locations of .NET such search may result in multiple possible locations.
+See [install-locations](https://github.com/dotnet/designs/blob/main/accepted/2020/install-locations.md)
+for more details. On Windows all of these locations are considered when looking
+for the right version of Runtime/SDK. This behavior is called "multi-level lookup".
+
+An example is if the machine has a private install in
+`C:\repo\.dotnet` (version 3.0) and a global install in
+`C:\Program Files\dotnet` (version 3.1).
+Running `C:\repro\.dotnet\dotnet.exe My30App.dll` will by default use
+the runtime version 3.1 from `C:\Program Files\dotnet`,
+even though it was started from the private install.
+
+This behavior can be disabled by setting `DOTNET_MULTILEVEL_LOOKUP=0`.
+
+Also of note is that this behavior is Windows specific, on non-Windows platforms
+only the first location is considered (the behavior is identical to `DOTNET_MULTILEVEL_LOOKUP=0`).
+
+## Feedback
+
+There have been a steady stream of feedback from all sides about this behavior:
+
+* It's often confusing: "I ran .NET from my private install, but it ended up
+using the global one anyway". Frequently users end up setting
+`DOTNET_MULTILEVEL_LOOKUP=0` globally on their machines to avoid this.
+* Inconsistent behavior between Windows and non-Windows.
+* Cause of breaks, typically in automated systems, where installing
+a new version of .NET globally changes behavior of otherwise isolated builds/tests.
+* Performance issues. These have shown up especially in CI environments
+(e.g. GitHub Actions) because a particular version of the SDK is downloaded,
+but a different version that happens to exist in the global location is used.
+Because these VMs are I/O constrained, they end up crossing the IOPS threshold
+and throttling, causing several second delays.
+
+## Disable this by default for .NET 6+ applications
+
+This is a proposal to change the default to disable multi-level lookup.
+
+* Applications will change behavior based on the target .NET version.
+This version will be determined by the `tfm` property in `.runtimeconfig.json`.
+    * Applications targeting .NET 6 and above will act as if `DOTNET_MULTILEVEL_LOOKUP`
+    is set to `0` by default.
+    * Applications targeting .NET 5 and below (.NET Core 3.\*, .NET Core 2.\*, ...)
+    will keep their current behavior which is as if `DOTNET_MULTILEVEL_LOOKUP`
+    is set to `1`.
+    * All applications will continue to obey an explicit setting of
```

Comment:
I think it would be better to always ignore `DOTNET_MULTILEVEL_LOOKUP` for .NET 6+ apps, with no option to override.

Reply:
I like this idea.  Are there any scenarios that we're aware of where MLL behavior is desired?

Reply:
Yup. I think this is the best plan.

Reply:
> Are there any scenarios that we're aware of where MLL behavior is desired?

Haven't really seen one.

Reply:
Thanks everybody - changed the doc to effectively remove MLL for TFM 7+.